Added the day8 assignment with tests

amankp-zop opened the pull request.

amankp-zop committed:
Added the day8 assignment with tests

aryanmehrotra commented on day8/handler/task/task_handler.go:
```diff
@@ -0,0 +1,127 @@
+	taskmodel "assignment8/models/task"
+)
+
+type TaskServicePort interface {
```
Interface can be named better

aryanmehrotra commented on day8/service/task/task_service.go:
```diff
@@ -0,0 +1,74 @@
+	if task.UserID == 0 {
+		return ErrUserIDRequired
+	}
+
```
These validations should be done in the handler layer, service layer is responsible about the business logic validations

amankp-zop committed:
added the gofr framework and implemented the test for it.

amankp-zop commented:
Implemented Swagger
Added the Gofr framework to the project.

Test Coverage:-
StoreLayer:- User: 95%, Task: 88%
ServiceLayer:- 100% both task and user
HandlerLayer:- 100% both task and user

aryanmehrotra commented on day10/main.go:
```diff
@@ -0,0 +1,62 @@
+	taskService := taskservice.NewTaskService(taskRepo)
+	taskHandler := taskhandler.NewTaskHandler(taskService)
+
+	mux := http.NewServeMux()
+	mux.Handle("/swagger/", httpSwagger.WrapHandler)
```
why do you need mux for when using gofr?

aryanmehrotra commented on day10/main.go:
```diff
@@ -0,0 +1,62 @@
+	app.PUT("/tasks/{id}/complete", taskHandler.MarkTaskComplete)
+	app.DELETE("/tasks/{id}", taskHandler.DeleteTask)
+
+	fmt.Println("Server running at http://localhost:8080")
```
gofr automatically logs why are we adding unnecessary logs?

aryanmehrotra commented on day10/handler/task/interface.go:
```diff
@@ -0,0 +1,15 @@
+	taskmodel "assignment8/models/task"
+)
+
+type TaskServicePort interface {
```
Is it as per the golang guidelines to name the interfaces?

aryanmehrotra commented on day10/handler/task/task_handler.go:
```diff
@@ -0,0 +1,96 @@
+
+func (h *TaskHandler) GetUserTasks(ctx *gofr.Context) (any, error) {
+	id, err := strconv.Atoi(ctx.PathParam("id"))
+	if err != nil {
+		return nil, err
+	}
```
Invalid param gofr errors should be returned as the status code would not be set correctly, it will be 500 but is it 500?

aryanmehrotra commented on day10/repository/task/task_repository.go:
```diff
@@ -0,0 +1,90 @@
+}
+
+func (*TaskRepository) CreateTask(c *gofr.Context, task *taskmodel.Task) error {
+	db := c.SQL
```
when we have to run just one query why to create a variable?

aryanmehrotra commented on day10/service/task/task_service.go:
```diff
@@ -0,0 +1,68 @@
+}
+
+func (s *TaskService) CreateTask(c *gofr.Context, task *taskmodel.Task) error {
+	if task.Title == "" {
+		return ErrTaskTitleRequired
+	}
+
+	if task.UserID == 0 {
+		return ErrUserIDRequired
+	}
+
+	return s.repo.CreateTask(c, task)
```
Dont we need to check if that user exists? by calling the service of user

aryanmehrotra commented on day10/service/task/task_service.go:
```diff
@@ -0,0 +1,68 @@
+}
+
+func (s *TaskService) GetTasksForUser(c *gofr.Context, userID int) ([]taskmodel.Task, error) {
+	if userID == 0 {
+		return nil, ErrUserIDInvalid
+	}
+
```
these typ pf basic validation should be done in handler

aryanmehrotra reviewed (changes requested):
These changes apply across the codebase

also, please create a differnet PR for day 10 tasks
